### ISSUE TITLE
WeKan Snap upgrade script, from 6.09 to newest

### ISSUE THREAD

**Issue by xet7** [wekan/wekan#5562]
Hi,
there is now upgrade script, that also migrates attachments and avatars.

At start of script is backup directory /var/wekanbackup , you may change it to directory where is enough disk space for copy of /var/snap/wekan/common and mongodump.

Check that you have enough disk space:
```
df -h
```
Check how much disk space /var/snap/wekan/common takes:
```
sudo du -sh /var/snap/wekan/common
```
If you need to attach external disk, that has more disk space for backup directory, some related info is at https://github.com/wekan/wekan/wiki/Repair-Docker

You can check what version you are running, with command:
```
sudo snap list
```

Example usage, at server, where is WeKan Snap 6.09:
```
sudo su

wget https://wekan.github.io/wekan-snap-upgrade.sh

chmod +x wekan-snap-upgrade.sh

./wekan-snap-upgrade.sh
```
Then, it will show menu:
```
Backups will be stored to /var/wekanbackup . Change it at top of this script, to where is enough disk space.
Backup directory requires a lot of disk space. It will have copy of /var/snap/wekan/common and mongodump.
1) Upgrade WeKan Snap from 6.09 to newest    3) Quit
2) Downgrade WeKan Snap from newest to 6.09
Please enter your choice:
```
When you write number `1`  and press Enter, upgrading starts.

If you need to downgrade, write `2`  and press Enter, downgrading starts.

If you write `3` and press Enter, it quits script.

Script will also show if your CPU does not support AVX, then at that old server new WeKan will not work, it requires moving WeKan to new server.

**Comment by xet7** [comment]
It looks like that to get the script to work, I had to manually first create backup directory:

```
sudo so

mkdir -p /var/wekanbackup
```

**Comment by Tanarri** [comment]
Hi,
I had tried you upgrade script as mentioned in Post 1.
It worked, but at the end I had following problems:
a) My Avatar was missing - no problem. I uploaded it again
b) the attachements are missing

A few Infos about a example attachment:
The script says:
![grafik](https://github.com/user-attachments/assets/ffa663a6-f7ec-4293-b421-9d7e7c807de4)

The filepath should be
![grafik](https://github.com/user-attachments/assets/5fb5cecc-0ce7-47f3-a7a5-edfd2e8e4fea)

The file ist missing:
![grafik](https://github.com/user-attachments/assets/543c39be-f9c5-4895-951d-75391a9652d9)

![grafik](https://github.com/user-attachments/assets/89295f12-d9d8-4808-b2fe-c8b8c49418ae)

The image should be here:
![grafik](https://github.com/user-attachments/assets/7568f46f-994c-4398-b970-9debf513e3b3)

Any hints or can I help you with more informations?

**Comment by xet7** [comment]
Upgrade script only works if it can rename database tables properly, cfs to attachments and avatars tables. If there is already existing tables, attachments and avatars are not visible, table rename does not work.

I try to rewrite upgrade script to export/import attachments, something like at https://github.com/wekan/minio-metadata

**Comment by lezioul** [comment]
> Hi, I had tried you upgrade script as mentioned in Post 1. It worked, but at the end I had following problems: a) My Avatar was missing - no problem. I uploaded it again b) the attachements are missing
> 
> A few Infos about a example attachment: The script says: ![grafik](https://private-user-images.githubusercontent.com/6181166/384897133-ffa663a6-f7ec-4293-b421-9d7e7c807de4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNDMwNTAsIm5iZiI6MTczMTM0Mjc1MCwicGF0aCI6Ii82MTgxMTY2LzM4NDg5NzEzMy1mZmE2NjNhNi1mN2VjLTQyOTMtYjQyMS05ZDdlN2M4MDdkZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTFUMTYzMjMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTQ1ZTY4NmUyNDQ1OTBlNjNlNzk2YmZmNTExZTk1ZjNjYWMzNmFkODkxMjA0NzQ2ZTg0MzQ0ODI4ODA1NmNhZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.WlaaalJ52xboVEvlqqc7yVaq4XwYS_WqOvVyoaOWmMk)
> 
> The filepath should be ![grafik](https://private-user-images.githubusercontent.com/6181166/384899771-5fb5cecc-0ce7-47f3-a7a5-edfd2e8e4fea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNDMwNTAsIm5iZiI6MTczMTM0Mjc1MCwicGF0aCI6Ii82MTgxMTY2LzM4NDg5OTc3MS01ZmI1Y2VjYy0wY2U3LTQ3ZjMtYTdhNS1lZGZkMmU4ZTRmZWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTFUMTYzMjMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzNiMDVlNTY0NjRmNzBmMzQzYTM5YmEwODA0ZjJhNjhmZTM3MDM1NzA5Zjg1OTZkZGZmMjY5ZTBjM2I2NWIyNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.2oMd17QQQ9GL9nBFqpYy_ahGUM5kPI-EG7fGbNEHdtg)
> 
> The file ist missing: ![grafik](https://private-user-images.githubusercontent.com/6181166/384898668-543c39be-f9c5-4895-951d-75391a9652d9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNDMwNTAsIm5iZiI6MTczMTM0Mjc1MCwicGF0aCI6Ii82MTgxMTY2LzM4NDg5ODY2OC01NDNjMzliZS1mOWM1LTQ4OTUtOTUxZC03NTM5MWE5NjUyZDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTFUMTYzMjMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzRiYzgxNWY4MTEwM2I1YTQyNTU2ZTBmMTM2NzA2NmM3OWM1NmQyNGFmNDUxZGFhZDY2ZjJjZmNkNzdiMWJkYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.j-WV3fsH0tMVzehKpzvOkpolz1zDZOrLkSo82T4oGIY)
> 
> ![grafik](https://private-user-images.githubusercontent.com/6181166/384897637-89295f12-d9d8-4808-b2fe-c8b8c49418ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNDMwNTAsIm5iZiI6MTczMTM0Mjc1MCwicGF0aCI6Ii82MTgxMTY2LzM4NDg5NzYzNy04OTI5NWYxMi1kOWQ4LTQ4MDgtYjJmZS1jOGI4YzQ5NDE4YWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTFUMTYzMjMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjM1ODY3YWRmNTU4MjA4ZDg2N2VkYTcwM2NiNTNkOTk0ZDkyN2Q3Yjc3NWY4YjMwOTcyNTgxYTZiYmFjMmM5NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.F33RSYCFnOF7QsuYj-bGpWuUxM_6bQT8OPenFzwo3m8)
> 
> The image should be here: ![grafik](https://private-user-images.githubusercontent.com/6181166/384897542-7568f46f-994c-4398-b970-9debf513e3b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNDMwNTEsIm5iZiI6MTczMTM0Mjc1MSwicGF0aCI6Ii82MTgxMTY2LzM4NDg5NzU0Mi03NTY4ZjQ2Zi05OTRjLTQzOTgtYjk3MC05ZGViZjUxM2UzYjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTFUMTYzMjMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjY2YTdiNWM4ZThjYTQ1NDYzNWQ2MGEwYmMwYzE3ZTUyNzM4OWJiNjhkZmQ2MjE3MzY3NjAzMTQ5MmE1N2JmNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.nUxBsZsWWJvKk3uX9FrhjeDdOJnqKQCIhWvtoVagEoI)
> 
> Any hints or can I help you with more informations?

Same experience here.



> Upgrade script only works if it can rename database tables properly, cfs to attachments and avatars tables. If there is already existing tables, attachments and avatars are not visible, table rename does not work.
> 
> I try to rewrite upgrade script to export/import attachments, something like at https://github.com/wekan/minio-metadata

what do you mean by : "If there is already existing tables, attachments and avatars are not visible, table rename does not work."? if i upgrade from stable to candidate, tables will be already existing, so what's the workaround?

**Comment by xet7** [comment]
TLDR: Wait for newer updated upgrade script. Use current upgrade script option to downgrade back to WeKan 6.09, until new upgrade script is available.

Long answer:

Old tables, where name starts with text `cfs`, from WeKan 6.09, do exist. But if some previous upgrade attempt, or uploading new images with WeKan 7.x has created tables that start with `Attachments` and `Avatars`, those new should be deleted, so that renaming `cfs` tables works. This existing upgrade script did work for someone already, where attachments and avatars become visible.

Also, some metadata is probably missing from avatars, that needs to be copied also, similar like metadata etc was copied for attachments. Those differences can be seen with MongoDB GUI like Nosqlbooster or DBGate.

There are still improvements to be made, because some images may not be detected correctly as image. For example, some `png` and other images may need `isImage: true` so that at opened card image preview is available, for example:

```
let pngResult = db.attachments.updateMany(
  {
    name: /.png$/i,
    isImage: {$ne: true}
  },
  {
    $set: {
    isImage: true,
    type: "image/png",
    mime: "image/png",
    "mime-type": "image/png"
    }
  }
);
```

**Comment by xet7** [comment]
Also, currently I'm figuring out, how to get MongoDB 6.x version that does not require CPU to support AVX, so that it would work at all hardware. https://github.com/wekan/wekan/issues/4321#issuecomment-2467933444

**Comment by ChristianAlfredsson** [comment]
This might solve it @xet7. Please try 

sudo snap run --shell wekan.mongodb
mongosh --port 27019
use wekan

db.attachments.files.drop()
db.attachments.chunks.drop()
db.avatars.files.drop()
db.avatars.chunks.drop()

db.cfs_gridfs.attachments.chunks.renameCollection('attachments.chunks')
db.cfs_gridfs.attachments.files.renameCollection('attachments.files')
db.cfs_gridfs.avatars.chunks.renameCollection('avatars.chunks')
db.cfs_gridfs.avatars.files.renameCollection('avatars.files')

db.attachments.updateMany(
  {
    name: /.png$/i,
    isImage: {$ne: true}
  },
  {
    $set: {
      isImage: true,
      type: "image/png",
      mime: "image/png",
      "mime-type": "image/png"
    }
  }
);

(Add more image types if it is necesary)

**Comment by lezioul** [comment]
> TLDR: Wait for newer updated upgrade script. Use current upgrade script option to downgrade back to WeKan 6.09, until new upgrade script is available.
> 
> Long answer:
> 
> Old tables, where name starts with text `cfs`, from WeKan 6.09, do exist. But if some previous upgrade attempt, or uploading new images with WeKan 7.x has created tables that start with `Attachments` and `Avatars`, those new should be deleted, so that renaming `cfs` tables works. This existing upgrade script did work for someone already, where attachments and avatars become visible.
> 
> Also, some metadata is probably missing from avatars, that needs to be copied also, similar like metadata etc was copied for attachments. Those differences can be seen with MongoDB GUI like Nosqlbooster or DBGate.
> 
> There are still improvements to be made, because some images may not be detected correctly as image. For example, some `png` and other images may need `isImage: true` so that at opened card image preview is available, for example:
> 
> ```
> let pngResult = db.attachments.updateMany(
>   {
>     name: /.png$/i,
>     isImage: {$ne: true}
>   },
>   {
>     $set: {
>     isImage: true,
>     type: "image/png",
>     mime: "image/png",
>     "mime-type": "image/png"
>     }
>   }
> );
> ```

feel free to ask me to test, i've a complete test env on a VM i can flush anytime :)

**Comment by ChristianAlfredsson** [comment]
Try this @lezioul 

1. Install 6.09
2. Add some cards with images and attachments. One image must be png
3. Upgrade with upgradescript
4. Runt this

sudo snap run --shell wekan.mongodb
mongosh --port 27019
use wekan

db.attachments.files.drop()
db.attachments.chunks.drop()
db.avatars.files.drop()
db.avatars.chunks.drop()

db.cfs_gridfs.attachments.chunks.renameCollection('attachments.chunks')
db.cfs_gridfs.attachments.files.renameCollection('attachments.files')
db.cfs_gridfs.avatars.chunks.renameCollection('avatars.chunks')
db.cfs_gridfs.avatars.files.renameCollection('avatars.files')

db.attachments.updateMany(
{
name: /.png$/i,
isImage: {$ne: true}
},
{
$set: {
isImage: true,
type: "image/png",
mime: "image/png",
"mime-type": "image/png"
}
}
);

Is attachments and png-images working?

**Comment by lezioul** [comment]
### What i've done : 

1. i install a fresh 6.09
2. import my dump
3. add a new card, take a snapshot, save it as a png file and attached it to the card
4. launch wekan-snap-upgrade.sh as root
5. use your commands : 

```
db.attachments.files.drop()
db.attachments.chunks.drop()
db.avatars.files.drop()
db.avatars.chunks.drop()
```
**-> false (each)**

```
db.cfs_gridfs.attachments.chunks.renameCollection('attachments.chunks')
db.cfs_gridfs.attachments.files.renameCollection('attachments.files')
db.cfs_gridfs.avatars.chunks.renameCollection('avatars.chunks')
db.cfs_gridfs.avatars.files.renameCollection('avatars.files')
```
**-> { ok: 1 } (for each)**

```
db.attachments.updateMany(
{
name: /.png$/i,
isImage: {$ne: true}
},
{
$set: {
isImage: true,
type: "image/png",
mime: "image/png",
"mime-type": "image/png"
}
}
);
```
->
**{
  acknowledged: true,
  insertedId: null,
  matchedCount: 18,
  modifiedCount: 18,
  upsertedCount: 0
}**

### Results : 

#### before update : 

![image](https://github.com/user-attachments/assets/b15b46c7-083d-4f39-8097-f4f31e2d9389)


#### after update : 

![image](https://github.com/user-attachments/assets/6405e7b0-0e18-40d5-943d-09a7982950d5)

**Comment by lezioul** [comment]
of course i did, with the right commands ;)

**Comment by hunter440p** [comment]
Hello 
I tried to install a fresh install of wekan in 6.09 (port : 3001), and then to upload an avatar + to create a card with an image attachment (with a copy paste directly)

After that, I tested the script and no errors were detected

But after the update :
- avatar doesn't show
- my card lost it's image
- i can't add new images (attachments shows : invalid-file ; if i add a new image, it doesn't show the image and the link is broken)
- The translations doesn't work 

My wekan used the port : 3001 before the migration.
After the migration, same port.

If I change the port to 80 (with the command : snap set wekan port='80'), the images are back, except for the avatar. The translations works again with port 80.

**Comment by MasterD0001** [comment]
Hi,
i have the same Problem using the Migration-Script. I'm on CentOS 7.9 and i had to change the pathes of snapd because it's under CentOS it is found under /var/lib/snapd/snap/wekan/current/ (or do i a think mistake?)

Whatever: If i change the Path the Script is running mostly successful, some failed to import:

Processing file with key: 5fd0e571761078179da549e0
Creating new format for: Anlage 1.pdf
Importing to new format...
2024-11-17T14:50:36.686+0100    connected to: mongodb://127.0.0.1:27019/
2024-11-17T14:50:36.687+0100    continuing through error: E11000 duplicate key error collection: wekan.attachments index: _id_ dup key: { _id: "5fd0e571761078179da549e0" }
2024-11-17T14:50:36.687+0100    0 document(s) imported successfully. 1 document(s) failed to import.
Completed migration for: Anlage 1.pdf

But also the successfully imported attachments aren't accessable from the GUI. It's like shown @lezioul. 
So we had to wait for an update for the script or is it a clear problem that i'm able to fix by myself?

Thanks again for your fast and usefull help

MasterD0001

**Comment by MasterD0001** [comment]
Hi there,

is there any Update in this case?
Thanks again for your great work

MasterdD0001

**Comment by xet7** [comment]
@MasterD0001

There will be updated script in near future, when I have tested current script, and improved or rewritten it. I will add comment to this issue when it's ready.

**Comment by lezioul** [comment]
@xet7 feel free to contact me if you want me to do some testing.

**Comment by Chralf** [comment]
I can also test. Very nice to get this to function.

**Comment by MasterD0001** [comment]
Hello again,

could you tell us some update about the new migration-script?
Thanks again

MasterD0001

**Comment by xet7** [comment]
@MasterD0001 

There will not be any separate migration script. There will only be new version of WeKan, that will work with any MongoDB database version, and any other database. 

Migrations at wekan/server/migrations.js will be removed, because doing any migrations (changes to database structure, or changes to boards, cards etc) takes too much time. Any existing data will be read directly from there were it currently is.

WeKan will directly read any existing data at any database version, and show that data.

For any migrations, if someone would like to migrate to different version of MongoDB, or migrate some other database than MongoDB, or migrate attachments from MongoDB to elsewhere, there will be options at Admin Panel to do migrations while WeKan is running. That will migrate all data, and change to use target database and files immediately after some text or file has been moved to new location.

Example how to detect version of MongoDB is here:

1) Javascript: Node.js and Bun:

https://github.com/wekan/ferretdb/blob/main/node-deno-bun/main.js#L20-L33

2) Javascript: Deno:

https://github.com/wekan/ferretdb/blob/main/node-deno-bun/main-deno.js#L20-L33

3) Go programming language:

https://github.com/wekan/wekango/blob/main/main.go#L39-L59

**Comment by MasterD0001** [comment]
Wow! Sounds really good. Looking forward for the new Version.  After that i'll try an Upgrade to a new version.
We have such the problem that the OS of our WeKan is running out of service. So we had to migratate/upgrade to another os.

**Comment by xet7** [comment]
Moved to #5694

**Comment by BinaryCraX** [comment]
I wanted to check on the current status of this. I noticed that issue #5694, which this was moved to, is now marked as completed. However, it's still unclear to me what the correct upgrade path is.

I ran the upgrade script today, but it failed to start MongoDB after the snap update. Could you clarify the recommended steps? Thank you very much in advance.

**Comment by xet7** [comment]
@BinaryCraX 

Correct upgrade path is to wait for upcoming automatic upgrade coming to Snap Stable channel. I'm currently developing that upgrade.

The problem in upgrades is, that attachments table names are different old vs new version. It's possible export/import attachments similar like scripts at https://github.com/wekan/minio-metadata , but I don't know does anyone dare to try to help.

There is:
- incomplete attachments upgrade code at https://github.com/wekan/wekan/blob/main/snap-src/bin/mongodb-control#L26-L82
- incomplete attachments migrate code that has problems with filenames with strange characters (some filenames are urlencoded, etc) at https://github.com/wekan/wekan/blob/main/server/migrations.js#L1250-L1445

That previous upgrade script tried to just rename tables to new names, but it did not work well enough.

Also, at Snap, it's about upgrading from MongoDB 3 to MongoDB 6, that may anyway need mongodump/mongorestore, or exporting text and attachments and importing back to newest version. But, for big databases, that may take many hours. That's why I try to add support for WeKan to work with old MongoDB version, without changing database version.

Current newest WeKan uses MongoDB 6. I'm also adding support for SQL databases, because MongoDB is not Open Source, and I'm working to make WeKan fully compliant with Standard for Public Code https://wekan.github.io/standard-for-public-code/

Anyway, these all are internal changes and upgrades to dependencies etc. WeKan UI and API stays same, and there will be more features. I'm developing upgrades for all WeKan platforms that are not yet using newest WeKan.

**Comment by FloRup** [comment]
Are there any updates? 

It's quite ridiculous how people that used snap are getting the short end of the stick and are pretty much forced to use an almost 3 year old version.

**Comment by Alingsaskommun** [comment]
We really love WeKan and we're so jealous of everyone running the newer versions! It's such a great system. Especially for us being in the EU where we have to keep all our data local.
WeKan lets us keep everything on our own servers which is perfect for all the rules we have to follow. It just works really well for what we need and we don't have to worry about where our data is going.
It sucks being stuck on 6.09 when we know there's so much cool stuff in the new versions, but honestly WeKan has been solid for us. Our team loves using it and we're just waiting for the day we can finally upgrade and get all those new features!

**Comment by xet7** [comment]
Both WeKan 6.09 and 7.x have bugs that are not possible to fix at architecture of those versions.

I'm developing a new WeKan version that will upgrade both of those to new WeKan version, that has new architecture.

**Comment by xet7** [comment]
When it's ready, for Snap, it will be released to all Snap Channels, like Stable etc.

**Comment by xet7** [comment]
I did give a talk about WeKan at Free Software Conferenfce Ukraine, there I mentioned about upcoming versions of WeKan:

https://github.com/wekan/wekan/releases/download/v7.90/2025-05-18_WeKan_at_Free_Software_Conference_Ukraine.webm

**Comment by wiking-at** [comment]
@xet7 - is the snap upgrade for version 6.0.9 still comming? or are the missing security fixes backported to 6.x branch?

**Comment by xet7** [comment]
@wiking-at 

Yes, upgrades are coming to all platforms: https://wekan.fi/upgrade/

**Comment by xet7** [comment]
@wiking-at 

> or are the missing security fixes backported to 6.x branch?

No.
